Added cleanup code for gcp resources

AnkitCLI opened the pull request:
This PR contains after hooks for test cases in which the after hook to clean up resources is not available.

itsankit-google reviewed (changes requested):
Why there are test failures with error in `Get count of no of records transferred to target BigQuery Table`:

```
om.google.cloud.bigquery.BigQueryException: Not found: Table cdf-athena:testing_bqmt.E2E_TARGET_cd33fe5f_1249_46b1_8fc3_b8da7dbce6f1 was not found in location US
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.translate(HttpBigQueryRpc.java:115)
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.queryRpc(HttpBigQueryRpc.java:652)
	at com.google.cloud.bigquery.BigQueryImpl$35.call(BigQueryImpl.java:1282)
	at com.google.cloud.bigquery.BigQueryImpl$35.call(BigQueryImpl.java:1279)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:105)
	at com.google.cloud.bigquery.BigQueryRetryHelper.run(BigQueryRetryHelper.java:64)
	at com.google.cloud.bigquery.BigQueryRetryHelper.runWithRetries(BigQueryRetryHelper.java:38)
	at com.google.cloud.bigquery.BigQueryImpl.queryRpc(BigQueryImpl.java:1278)
	at com.google.cloud.bigquery.BigQueryImpl.query(BigQueryImpl.java:1266)
	at io.cdap.e2e.utils.BigQueryClient.getSoleQueryResult(BigQueryClient.java:86)
	at io.cdap.e2e.utils.BigQueryClient.countBqQuery(BigQueryClient.java:46)
	at io.cdap.plugin.bigquery.stepsdesign.BigQueryBase.getCountOfNoOfRecordsTransferredToTargetBigQueryTable(BigQueryBase.java:90)
	at ✽.Get count of no of records transferred to target BigQuery Table(file:src/e2e-test/features/bigquery/sink/GCSToBigQuery.feature:39)
Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 404 Not Found
POST https://www.googleapis.com/bigquery/v2/projects/cdf-athena/queries
{
  "code" : 404,
  "errors" : [ {
    "domain" : "global",
    "message" : "Not found: Table cdf-athena:testing_bqmt.E2E_TARGET_cd33fe5f_1249_46b1_8fc3_b8da7dbce6f1 was not found in location US",
    "reason" : "notFound"
  } ],
  "message" : "Not found: Table cdf-athena:testing_bqmt.E2E_TARGET_cd33fe5f_1249_46b1_8fc3_b8da7dbce6f1 was not found in location US",
  "status" : "NOT_FOUND"
```

itsankit-google commented:
> Why there are test failures with error in `Get count of no of records transferred to target BigQuery Table`:
> 
> ```
> om.google.cloud.bigquery.BigQueryException: Not found: Table cdf-athena:testing_bqmt.E2E_TARGET_cd33fe5f_1249_46b1_8fc3_b8da7dbce6f1 was not found in location US
> 	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.translate(HttpBigQueryRpc.java:115)
> 	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.queryRpc(HttpBigQueryRpc.java:652)
> 	at com.google.cloud.bigquery.BigQueryImpl$35.call(BigQueryImpl.java:1282)
> 	at com.google.cloud.bigquery.BigQueryImpl$35.call(BigQueryImpl.java:1279)
> 	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:105)
> 	at com.google.cloud.bigquery.BigQueryRetryHelper.run(BigQueryRetryHelper.java:64)
> 	at com.google.cloud.bigquery.BigQueryRetryHelper.runWithRetries(BigQueryRetryHelper.java:38)
> 	at com.google.cloud.bigquery.BigQueryImpl.queryRpc(BigQueryImpl.java:1278)
> 	at com.google.cloud.bigquery.BigQueryImpl.query(BigQueryImpl.java:1266)
> 	at io.cdap.e2e.utils.BigQueryClient.getSoleQueryResult(BigQueryClient.java:86)
> 	at io.cdap.e2e.utils.BigQueryClient.countBqQuery(BigQueryClient.java:46)
> 	at io.cdap.plugin.bigquery.stepsdesign.BigQueryBase.getCountOfNoOfRecordsTransferredToTargetBigQueryTable(BigQueryBase.java:90)
> 	at ✽.Get count of no of records transferred to target BigQuery Table(file:src/e2e-test/features/bigquery/sink/GCSToBigQuery.feature:39)
> Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 404 Not Found
> POST https://www.googleapis.com/bigquery/v2/projects/cdf-athena/queries
> {
>   "code" : 404,
>   "errors" : [ {
>     "domain" : "global",
>     "message" : "Not found: Table cdf-athena:testing_bqmt.E2E_TARGET_cd33fe5f_1249_46b1_8fc3_b8da7dbce6f1 was not found in location US",
>     "reason" : "notFound"
>   } ],
>   "message" : "Not found: Table cdf-athena:testing_bqmt.E2E_TARGET_cd33fe5f_1249_46b1_8fc3_b8da7dbce6f1 was not found in location US",
>   "status" : "NOT_FOUND"
> ```

Fixed by https://github.com/cdapio/cdap/pull/15724.

itsankit-google commented on src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:
```diff
@@ -1636,4 +1640,12 @@ public static void deleteTargetBqmtTable() throws IOException, InterruptedExcept
     }
   }
+
+  @After(order = 1, value = "@BQ_SECOND_RECORD_SOURCE_TEST")
```
Should this be BQ_RECORD_SOURCE_TEST? I don't see this tag being used anywhere.

AnkitCLI replied:
we are using this is in bqtobq Additional.feature file. so **BQ_SECOND_RECORD_SOURCE_TEST** is used here

itsankit-google replied:
I see, can we please rename both these tags to something meaningful?

It is hard to understand what these both tags are referring to `BQ_RECORD_SOURCE_TEST` & `BQ_SECOND_RECORD_SOURCE_TEST`?

AnkitCLI replied:
changed done.

itsankit-google commented on src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:
```diff
@@ -1636,4 +1640,12 @@ public static void deleteTargetBqmtTable() throws IOException, InterruptedExcept
+  @After(order = 1, value = "@BQ_SECONDARY_RECORD_SOURCE_TEST")
+  public static void deleteTempSource2BQTable() throws IOException, InterruptedException {
+    BigQueryClient.dropBqQuery(bqSourceTable2);
+    PluginPropertyUtils.removePluginProp("bqSourceTable2");
```
before removing the plugin property, we need to fetch it for logging right?

Otherwise `bqSourceTable2` would be empty?

Shouldn't it be something like:
```
bqSourceTable2 = PluginPropertyUtils.pluginProp("bqSourceTable2");
PluginPropertyUtils.removePluginProp("bqSourceTable2");
```

AnkitCLI replied:
Actually the logger is at right position. The way you have written is also right but we already assigned globally assigned **bqSourceTable2** in before hook. so no need to add in after hook. And either after removing PluginPropertyUtils.removePluginProp("bqSourceTable2") we still have globally assigned **bqSourceTable2** so we can use logger after remove step.

itsankit-google replied:
The source of truth here would be the `pluginProperties` file so I think we should always fetch the values from there instead of relying on the before hook.

AnkitCLI replied:
added the step

itsankit-google reviewed (commented):
Similar comment in other tests as well, for example: 
![image](https://github.com/user-attachments/assets/6b2565b7-874d-45ea-9f8e-2ae8cb90b011)

https://github.com/data-integrations/google-cloud/pull/1453/files#diff-d7999135c7a28d37bb9cad09ae4452055ce57b2f8fcfdaf123053690205b6cdeR1613-R1614

AnkitCLI commented:
> Similar comment in other tests as well, for example: ![image](https://private-user-images.githubusercontent.com/88528384/379105363-6b2565b7-874d-45ea-9f8e-2ae8cb90b011.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk2NjA5NDAsIm5iZiI6MTcyOTY2MDY0MCwicGF0aCI6Ii84ODUyODM4NC8zNzkxMDUzNjMtNmIyNTY1YjctODc0ZC00NWVhLTlmOGUtMmFlOGNiOTBiMDExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIzVDA1MTcyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkNzJlZWNkNGRjNjMyZGM3ZjdmYjRlNjVhNGEzOWZkY2JhMTE5ZmZlYjc3OTQ5OGQyZDZmYTIwYzE3ZDc5ZWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NxtKVEWfjFqfnBnGhXVrtuuTVtr8kCxJEVaT5ztXkbc)
> 
> https://github.com/data-integrations/google-cloud/pull/1453/files#diff-d7999135c7a28d37bb9cad09ae4452055ce57b2f8fcfdaf123053690205b6cdeR1613-R1614
As in this code we are not using globally assigned target tables we are just using table names that we provided in plugin parameters file so that's it is not giving the target table names in this one.

itsankit-google commented:
> As in this code we are not using globally assigned target tables we are just using table names that we provided in plugin parameters file so that's it is not giving the target table names in this one.

Even though plugin automatically creating the tables while writing, we do need to clean them up, and we should also correctly log which tables are getting deleted.

AnkitCLI commented:
> > As in this code we are not using globally assigned target tables we are just using table names that we provided in plugin parameters file so that's it is not giving the target table names in this one.
> 
> Even though plugin automatically creating the tables while writing, we do need to clean them up, and we should also correctly log which tables are getting deleted.

added the logger step and removed the step for clean up
